### PR TITLE
Grouped Items on merchant Item index page by status (disabled/enabled) with button to change status

### LINKED ISSUE #31: Merchant Items Grouped by Status
As a merchant,
When I visit my merchant items index page
Then I see two sections, one for "Enabled Items" and one for "Disabled Items"
And I see that each Item is listed in the appropriate section

### PR DESCRIPTION
**Pull request by kbhoffmann**
Created a method in merchant model to filer by item status.  Method called in Merchant Items index page. Updated spec file and view to make sections for disabled and enabled items.  Adding heading for page.

### CODE REVIEW

File: app/views/merchant_items/index.html.erb [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-      <%= button_to "Disable" ,"/merchants/#{@merchant.id}/items/#{item.id}", method: :patch, local: true, params: {status: "disabled"} %>
-    <% end  %>
-  </div>
-<% end %>
+<h1>Items for Merchant: <%= @merchant.name %></h1>
+
+<div id="Enabled-items">
+  <h3>Enabled Items</h3>
+  <% @merchant.filter_item_status(1).each do |item| %>
+    <%= link_to "#{item.name}", "/merchants/#{@merchant.id}/items/#{item.id}" %><%= button_to "Disable" ,"/merchants/#{@merchant.id}/items/#{item.id}", method: :patch, local: true, params: {status: "disabled"} %>
```
Comment (joel-grant) [inline app/views/merchant_items/index.html.erb:6]: I like your use of the extra params in your button.  I thought this section in general was nicely done.  All the work you did looks appropriate for a View and is easy to understand what you want from it. I also liked that you added `<div>` to take advantage of within blocks for your tests

File: spec/models/merchant_spec.rb [hunk 2]
```diff
@@ -74,4 +74,17 @@
   describe "validations" do
     it { should validate_presence_of(:name) }
   end
+
+  describe 'instance variables' do
+    it 'filters items by status' do
+      merchant_1 = FactoryBot.create(:merchant)
+      item_1 = FactoryBot.create(:item, status: 0, merchant: merchant_1)
```
Comment (joel-grant) [inline spec/models/merchant_spec.rb:81]: Teach me your ways!  Can't wait to get factorybot into my own tests.  Your's are clean and crisp

**Review by joel-grant — APPROVED** [review]

File: spec/features/merchants/items/index_spec.rb [hunk 3]
```diff
@@ -36,26 +36,21 @@
...
+      within("#Disabled-items") do
+        expect(item_1.status).to eq("disabled")
+        first(:button, "Enable").click
+        expect(current_path).to eq("/merchants/#{merch_1.id}/items")
+        updated_item_1 = Item.find(item_1.id)
+        expect(updated_item_1.status).to eq("enabled")
+      end
     end
   end
 end
```
Comment (ParkerLockhart) [inline spec/features/merchants/items/index_spec.rb:56]: Oooh I'm gonna have to take a look at my enable/disable tests too after I implement this functionality on admin merchants

**Review by ParkerLockhart — APPROVED** [review]
Heck yes! This is looking really cool.

**Pull request merged**